libs/filer/workspace_files_extensions_client.go

```diff
@@ -301,6 +312,22 @@ func (w *workspaceFilesExtensionsClient) Delete(ctx context.Context, name string
 func (w *workspaceFilesExtensionsClient) Stat(ctx context.Context, name string) (fs.FileInfo, error) {
 	info, err := w.wsfs.Stat(ctx, name)
 
+	// If an object is found and it is not a notebook, return the stat object. This check is done
+	// to avoid returning the stat for a notebook called `foo` when the user actually
+	// wanted to stat a file called `foo`.
+	//
+	// To stat the metadata of a notebook called `foo` in the workspace the user
+	// should use the name with the extension included like `foo.ipynb` or `foo.sql`.
+	if err == nil && info.Sys().(workspace.ObjectInfo).ObjectType != workspace.ObjectTypeNotebook {
+		return info, nil
+	}
+
+	// If a notebook is found by the workspace files client, without having stripped
+	// the extension, this implies that no file with the same name exists.
+	if err == nil && info.Sys().(workspace.ObjectInfo).ObjectType == workspace.ObjectTypeNotebook {
+		return nil, FileDoesNotExistError{name}
+	}
+
```

Comment:
These are the success cases, please move below.

It can catch any non-nil err and directly return it as well.

Reply:
Good point, done!